### PR TITLE
sage: downgrade maxima to 5.41

### PR DESCRIPTION
**Pull request by timokau**
###### Motivation for this change

See discussion at https://github.com/NixOS/nixpkgs/commit/82254747af35f3e0e0d6f78023ded3a81e25331b.

CC @peti @7c6f434c 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by 7c6f434c — APPROVED** [review]

File: pkgs/applications/science/math/maxima/default.nix [hunk 1]
```diff
@@ -70,10 +70,6 @@ stdenv.mkDerivation ({
       url = "https://git.sagemath.org/sage.git/plain/build/pkgs/maxima/patches/maxima.system.patch?id=07d6c37d18811e2b377a9689790a7c5e24da16ba";
       sha256 = "18zafig8vflhkr80jq2ivk46k92dkszqlyq8cfmj0b2vcfjwwbar";
     })
-    # There are some transient test failures. I hope this disables all those tests.
-    # If those test failures ever happen in the non-ecl version, that should be
-    # reportetd upstream.
-    ./known-ecl-failures.patch
```
Comment (peti) [inline pkgs/applications/science/math/maxima/default.nix]: Why did you remove that patch?

Comment (7c6f434c) [reply pkgs/applications/science/math/maxima/default.nix]: My impression: ECL Maxima is only worth the effort with Sage, so the ECL-specific code can be just present in the version used for Sage. As upstream changes in both Maxima and Sage (that might settle on some patches for Maxima before it is released upstream) happen, we can react to them accordingly.

Comment (peti) [reply pkgs/applications/science/math/maxima/default.nix]: I see. That makes sense.

Comment (timokau) [reply pkgs/applications/science/math/maxima/default.nix]: After the update the patch didn't apply anymore, which broke `maxima-ecl`. Since all tests are now disabled anyway, I saw little point in putting in the effort of rebasing it.
